libbpf-cargo/src/btf/btf.rs

```diff
@@ -381,26 +412,85 @@ impl<'a> Btf<'a> {
 
                             if padding != 0 {
                                 writeln!(
-                                    def,
+                                    type_content,
                                     r#"    __pad_{offset}: [u8; {padding}],"#,
                                     offset = offset,
                                     padding = padding,
                                 )?;
+
+                                impl_default.push(format!(
+                                    r#"__pad_{offset}: [u8::default(); {padding}]"#,
+                                    offset = offset,
+                                    padding = padding,
+                                ));
                             }
                         }
 
                         // Set `offset` to end of current var
                         offset = ((member.bit_offset / 8) + self.size_of(field_ty_id)?) as usize;
 
+                        if can_default_impl {
+                            let fstripped_type_id = self.skip_mods_and_typedefs(field_ty_id)?;
+                            let fty = self.type_by_id(fstripped_type_id)?;
+                            match fty {
+                                BtfType::Array(ft) => {
+                                    if ft.nelems > 32 {
+                                        default_macro = false
+                                    }
+                                }
+                                _ => {},
+                            }
+
+                            let field_ty_def = match self.type_default(field_ty_id) {
+                                Ok(def) => def,
+                                Err(_) => {
+                                    can_default_impl = false;
+                                    String::from("")
+                                }
+                            };
```

Comment:
Doesn't an error from `self.type_default()` imply malformed BTF? I think this can just be:
```
let field_ty_def = self.type_default(field_ty_id)?;
```
and we can have one less variable (`can_default_impl`)